backend/prisma/schema.prisma

```diff
@@ -8,54 +8,52 @@ datasource db {
 }
 
 model User {
-  id Int @id @default(autoincrement())
-  username String @unique
-  email String? @unique
-  password String?
-  fullName String
-  imageUrl String?
-  files File[]
-  sentChats Chat[] @relation("SentChats")
-  receivedChats Chat[] @relation("ReceivedChats")
-  chatbotInteraction ChatbotInteraction[]
-  peers Peer[] @relation("UserPeers")
-  peerOf Peer[] @relation("PeerUsers")
+  id               Int           @id @default(autoincrement())
+  username         String        @unique
+  email            String?       @unique
+  password         String?
+  fullName         String
+  imageUrl         String?
+  createdAt        DateTime      @default(now())
+  updatedAt        DateTime      @default(now()) @updatedAt
+  codeGroups       CodeGroup[]   @relation("UserCodeGroups")
+  groupMemberships GroupMember[] @relation("UserGroupMemberships")
+  addedGroups      GroupMember[] @relation("AddedByUser")
+  files            File[]        @relation("UserFiles")
 }
 
-model File {
-  id Int @id @default(autoincrement())
-  fileName String
-  fileContent String
-  userId Int
-  user User @relation(fields: [userId], references: [id])
-  createdAt DateTime @default(now())
-  updatedAt DateTime @updatedAt
+model CodeGroup {
+  id        Int           @id @default(autoincrement())
+  groupName String
+  imgUrl    String?
+  creatorId Int
+  createdAt DateTime      @default(now())
+  creator   User          @relation("UserCodeGroups", fields: [creatorId], references: [id])
+  members   GroupMember[] @relation("GroupMembers")
+  files     File[]        @relation("GroupFiles")
 }
 
-model Chat {
-  id Int @id @default(autoincrement())
-  message String
-  fromUserId Int
-  toUserId Int
-  fromUser User @relation("SentChats", fields: [fromUserId], references: [id])
-  toUser User @relation("ReceivedChats", fields: [toUserId], references: [id])
-  createdAt DateTime @default(now())gidvjcrdlkkekfkkfbbvdjchnnrveunr
-}
+model GroupMember {
```

Comment:
Code groups are only the groups created by the user
so, group member model is used as:
1. Sub model for the members field in Code Group
2. Sub model for memberships to show which group the user was added to
3. Sub model for addedGroups to show how many users the user has added across all groups
![Screenshot](https://github.com/izuchukwumba/peered/assets/45977460/1cd089c1-9af7-450b-b1aa-fea177474038)

Reply:
I see, for the groups a user was added to, what was your reasoning for representing that with the GroupMember model, instead of the CodeGroup model, like is used for groups a user created?

Reply:
Also how do you plan to use the addedGroups field? Wondering if this field is necessary